File: 0028-find-the-index-of-the-first-occurrence-in-a-string/0028-find-the-index-of-the-first-occurrence-in-a-string.java
class Solution {
    public int strStr(String haystack, String needle) {
        // return(haystack.indexOf(needle));
        int h = haystack.length(), n = needle.length(), index = -1, l = 0, m = 0, j;
        // boolean occurance=flase;
        while (l <= h - n) {
            m = 0;
            if (haystack.charAt(l) == needle.charAt(0)) {
                index = l;
                j = l + 1;
                while (j < l + n) {
                    m++;
                    if (haystack.charAt(j) != needle.charAt(m)) {
                        index = -1;
                        break;
                    }
                    j++;
                }
                if (index != -1) {
                    return index;
                }
            }

            l++;
        }
        return index;
    }
}

File: 0028-find-the-index-of-the-first-occurrence-in-a-string/0028-find-the-index-of-the-first-occurrence-in-a-string.py
class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        index =-1
        
        hay_length=len(haystack)
        nee_length=len(needle)
        if hay_length < nee_length :
            return index

        
        for i in range (hay_length-nee_length+1):
            j=0
            
            while j<nee_length :
                
                    if  haystack[i+j]==needle[j] :
                    #   print(haystack[k],needle[j])
                      
                      index=i
                    else:
                        index=-1
                        break
                    j=j+1

            if index != -1 :
                return index
        
            

        return index